### PR TITLE
ClientTheme light mode support, shortcut buttons

### PR DESCRIPTION
**Pull request by CodeF53**
Pretty confident this is up to Vencord standard

### CODE REVIEW

File: src/plugins/clientTheme/clientTheme.css [hunk 1]
```diff
@@ -22,3 +22,17 @@
 .client-theme-warning {
     color: var(--text-danger);
 }
+
+/* color fixes for light mode */
+.theme-light {
```
Comment (Vendicated) [inline src/plugins/clientTheme/clientTheme.css:27]: you're changing global styles in a non managed css file. thats baaaaad. make your css managed

File: src/plugins/clientTheme/index.tsx [hunk 2]
```diff
@@ -31,9 +32,26 @@ function onPickColor(color: number) {
     updateColorVars(hexColor);
 }
 
+const saveClientTheme = findByPropsLazy("saveClientTheme");
```
Comment (Vendicated) [inline src/plugins/clientTheme/index.tsx:35]:
```suggestion
const { saveClientTheme } = findByPropsLazy("saveClientTheme");
```

File: src/plugins/clientTheme/index.tsx [hunk 3]
```diff
@@ -30,9 +34,31 @@ function onPickColor(color: number) {
     updateColorVars(hexColor);
 }
 
+const { saveClientTheme } = findByPropsLazy("saveClientTheme");
+
+function swapTheme() {
```
Comment (Nuckyz) [inline src/plugins/clientTheme/index.tsx:39]:
```suggestion
function swapTheme(currentTheme: string) {
```

File: src/plugins/clientTheme/index.tsx [hunk 4]
```diff
@@ -30,9 +34,31 @@ function onPickColor(color: number) {
...
+    const isLightTheme = getTheme() === Theme.Light;
```
Comment (Nuckyz) [inline src/plugins/clientTheme/index.tsx:40]:
```suggestion
```

File: src/plugins/clientTheme/index.tsx [hunk 5]
```diff
@@ -30,9 +34,31 @@ function onPickColor(color: number) {
...
+    saveClientTheme({ theme: isLightTheme ? "dark" : "light" });
```
Comment (Nuckyz) [inline src/plugins/clientTheme/index.tsx:41]:
```suggestion
    saveClientTheme({ theme: currentTheme === "light" ? "dark" : "light" });
```

File: src/plugins/clientTheme/index.tsx [hunk 6]
```diff
@@ -48,12 +74,12 @@ function ThemeSettings() {
...
             </div>
-            {lightnessWarning || lightModeWarning
+            {contrastWarning
                 ? <div>
                     <Forms.FormDivider className={classes(Margins.top8, Margins.bottom8)} />
                     <Forms.FormText className="client-theme-warning">Your theme won't look good:</Forms.FormText>
-                    {lightnessWarning && <Forms.FormText className="client-theme-warning">Selected color is very light</Forms.FormText>}
-                    {lightModeWarning && <Forms.FormText className="client-theme-warning">Light mode isn't supported</Forms.FormText>}
+                    <Forms.FormText className="client-theme-warning">Selected color won't contrast well with text</Forms.FormText>
+                    {!unFixableContrast && <Button onClick={swapTheme}>Swap Theme</Button>}
```
Comment (Nuckyz) [inline src/plugins/clientTheme/index.tsx:82]:
```suggestion
                    {!unFixableContrast && <Button onClick={() => swapTheme(theme)}>Swap Theme</Button>}
```

**Review by Nuckyz — CHANGES_REQUESTED** [review]

**Comment by CodeF53** [comment]
Last commit was pretty big, as it added another generator

I switched to generating light mode fixing css because:
1. Overriding white-500 screwed with font color rendering
2. Manually writing overwrites for every single usage of white-500 for a background is not fun. (the generator makes 80 lines of css)
3. Manually written overrides will break, and a major part of the plugin is to generate css so it doesn't break when discord changes/adds/removes stuff.

Also in this commit, lots of code was moved to helper functions due to usage by both `generateColorOffsets` and `generateLightModeFixes`

**Comment by CodeF53** [comment]
Its stuck saying there is a "Changes requested 1 review requesting changes" when I have already resolved everything

File: src/plugins/clientTheme/index.tsx [hunk 7]
```diff
@@ -8,19 +8,19 @@ import "./clientTheme.css";
...
 import { definePluginSettings } from "@api/Settings";
 import { Devs } from "@utils/constants";
-import { getTheme, Theme } from "@utils/discord";
 import { Margins } from "@utils/margins";
 import { classes } from "@utils/misc";
 import definePlugin, { OptionType, StartAt } from "@utils/types";
-import { findComponentByCodeLazy } from "@webpack";
-import { Button, Forms } from "@webpack/common";
+import { findByPropsLazy, findComponentByCodeLazy, findStoreLazy } from "@webpack";
+import { Button, Forms, lodash as _, useStateFromStores } from "@webpack/common";
```
Comment (FieryFlames) [inline src/plugins/clientTheme/index.tsx:15]: why import lodash as _?

File: src/plugins/clientTheme/index.tsx [hunk 8]
```diff
@@ -30,9 +30,36 @@ function onPickColor(color: number) {
...
-    const lightModeWarning = getTheme() === Theme.Light;
+    const theme = useStateFromStores([ThemeStore], () => ThemeStore.theme);
+    const isLightTheme = theme === "light";
+    const oppositeTheme = isLightTheme ? "dark" : "light";
+
+    const nitroTheme = useStateFromStores([NitroThemeStore], () => NitroThemeStore.gradientPreset);
+    const nitroThemeEnabled = nitroTheme !== undefined;
+
+    const selectedLuminance = relativeLuminance(settings.store.color);
+    console.log(selectedLuminance);
```
Comment (FieryFlames) [inline src/plugins/clientTheme/index.tsx:51]: a console.log 😨

**Pull request merged**